broker-daemon/relayer/identity.js

```diff
@@ -70,37 +53,28 @@ class Identity {
   }
 
   /**
-   * @typedef {Object} Authorization
-   * @property {String} timestamp Int64 string of the current unix timestamp in seconds
-   * @property {String} nonce base64 string of 32 random bytes
-   * @property {String} signature base64 string of the signature that authorizes it
-   */
-
-  /**
-   * Authorize a request to act on a given object by signing its id
+   * Authorize a gRPC Request by adding metadata to it.
    *
    * Specifically, the signature is of the following payload:
    *  - timestamp of the request, in seconds. A timestamp within +/- 60 seconds of the Relayer's time should be accepted.
    *  - A random nonce of 32 bytes, represented in base64. This nonce should not be repeated, but the Relayer may not reject duplicate nonces older than 24 hours.
-   *  - Id of the object to authorize access to
    *
    * This payload is joined by commas (','), and signed using the public key of the broker.
    *
    * The request can then be validated by the Relayer as being genuine from the owner of the public key.
-   * @param  {String} id - id to authorize a request for
-   * @return {Authorization}
+   * @return {grpc.Metadata} Metadata object with necessary items to verify it
    */
-  authorize (id) {
+  authorize () {
     const timestamp = nowInSeconds().toString()
     const nonce = randomBytes(32).toString('base64')
-    const payload = [ timestamp, nonce, id ].join(',')
+    const payload = [ timestamp, nonce ].join(',')
     const signature = this.sign(payload)
+    const pubKey = this.pubKeyBase64
 
-    return {
-      timestamp,
-      nonce,
-      signature
-    }
+    const metadata = new Metadata()
+    metadata.set('authorization', `${pubKey},${timestamp},${nonce},${signature}`)
```

Comment:
did you not want to set pubkey/timestamp/nonce as separate fields for metadata? Then we wouldn't have to parse this information (via string) on the server